packages/neon_framework/lib/src/widgets/dialog.dart

```diff
@@ -523,3 +555,395 @@ class NeonUnifiedPushDialog extends StatelessWidget {
+        );
+      },
+    );
+  }
+
+  Widget _buildPredefinedStatuses() => ResultBuilder.behaviorSubject(
+        subject: bloc.predefinedStatuses,
+        builder: (final context, final predefinedStatusesResult) => ResultBuilder.behaviorSubject(
+          subject: bloc.status,
+          builder: (final context, final statusResult) {
+            final selectedPredefinedStatus = statusResult.hasData && statusResult.requireData is user_status.Private
+                ? (statusResult.requireData as user_status.Private).messageId
+                : null;
+
+            final tiles = predefinedStatusesResult.data
+                ?.map(
+                  (final status) => Card(
+                    shape: selectedPredefinedStatus == status.id ? selectedBorder : null,
+                    child: InkWell(
+                      borderRadius: const BorderRadius.all(Radius.circular(12)),
+                      onTap: () {
+                        bloc.setPredefinedMessage(
+                          id: status.id,
+                          clearAt: status.clearAt != null
+                              ? status.clearAt.toDateTime()!.millisecondsSinceEpoch ~/ 1000
+                              : null,
+                        );
+                      },
+                      child: ListTile(
+                        leading: Text(
+                          status.icon,
+                          style: const TextStyle(fontSize: 18),
+                        ),
+                        title: Text(status.message),
+                        subtitle: Text(status.clearAt.formatRelative(context)),
+                      ),
+                    ),
+                  ),
+                )
+                .toList();
+
+            return Column(
+              children: [
+                NeonError(
+                  predefinedStatusesResult.error,
+                  onRetry: bloc.refresh,
+                ),
+                NeonLinearProgressIndicator(
+                  visible: predefinedStatusesResult.isLoading,
+                ),
+                if (tiles != null) buildGrid(NeonDialogTheme.of(context).constraints, tiles),
+              ],
+            );
+          },
+        ),
+      );
+
+  ShapeBorder get selectedBorder => RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(12),
+        side: BorderSide(
+          color: Theme.of(context).colorScheme.primary,
+          width: 1.5,
+        ),
+      );
+
+  Widget buildGrid(
+    final BoxConstraints constraints,
+    final List<Widget> children,
+  ) =>
+      SizedBox(
+        height: 72 * 3, // TODO: This needs to go away to allow dynamic sizing.
+        child: Wrap(
+          children: children
+              .map(
+                (final child) => SizedBox(
+                  height: 72,
+                  width: constraints.maxWidth / 2,
+                  child: child,
+                ),
+              )
+              .toList(),
+        ),
+      );
```

Comment:
```suggestion
  Widget buildGrid(
    final BoxConstraints constraints,
    final List<Widget> children,
  ) =>
      Flexible(
        child: Wrap(
          children: children
              .map(
                (final child) => SizedBox(
                  height: 72,
                  width: constraints.maxWidth / 2,
                  child: child,
                ),
              )
              .toList(),
        ),
      );
```

Does this really simple trick work?

Reply:
Nope, already tried it and just checked again.

Reply:
Then I don't really understand what you want.
Can you describe it? or two mockups for different screen sizes (figma/penpot?)

Reply:
It is quite simple: If there is enought horizontal screen space then the tiles should be displayed in two columns (achieved by setting the width to half of the max width). If the space is not enough then it should fallback to a single column (which is done by the Wrap automatically).

Reply:
And the wrapped tile should take the entire width of the dialog or still the same size as if it wasn't collapsed?

If it should resize the solution is a GridView if it should stay the same size (which I find more pleasant when resizing on the desktop the above solution should work (I'll double check with my local changes but I got it to work yesterday).

Reply:
Yeah it should stay the same size. Please tell me how you got it working :D

Reply:
My bad you'll also need to remove the nested Columns that are used for the loading and error handling (I dislike this pattern so much that I immediately changed it before trying the Flexible but forgot to mention it yesterday).

Reply:
Should I do a complete review as there are also other code changes that I'd like to be made?

Reply:
I haven't written any tests yet so that is still to follow, but you can review the existing code if you like to do it already.